### PR TITLE
Xdoctest

### PR DESCRIPTION
**Pull request by yuchen202**
### PR types

 Others

### PR changes

 Others 

### Description

我只会改这些了。。。有啥具体的问题还请孙师傅指导一下
关联PR：https://github.com/PaddlePaddle/Paddle/pull/55933

### CODE REVIEW

File: python/paddle/nn/functional/activation.py [hunk 1]
```diff
@@ -49,12 +49,13 @@ def celu(x, alpha=1.0, name=None):
...
-            out = F.celu(x, alpha=0.2)
-            # [[-0.19865242,  6.        ],
-            #  [ 1.        , 15.60000038]]
+            >>> import paddle
+            >>> import paddle.nn.functional as F
+
+            >>> x = paddle.to_tensor([[-1., 6.], [1., 15.6]])
+            >>> out = F.celu(x, alpha=0.2)
+                [[-0.19865242,  6.        ],
+                [ 1.        , 15.60000038]]
```
Comment (sunzhongkai588) [inline python/paddle/nn/functional/activation.py]:
```suggestion
            >>> x = paddle.to_tensor([[-1., 6.], [1., 15.6]])
            >>> out = F.celu(x, alpha=0.2)
            >>> print(out)
            [[-0.19865242,  6.        ],
             [ 1.        , 15.60000038]]
```

- 代码示例如果有输出，应该加 `print()`  或者直接写变量名 `out`

- 输出的内容的缩进应该和 `>>>` 对齐

File: python/paddle/nn/functional/activation.py [hunk 2]
```diff
@@ -1727,18 +1733,18 @@ def gumbel_softmax(x, temperature=1.0, hard=False, axis=-1, name=None):
...
+
+            >>> logits = paddle.randn([4, 6])
+            >>> temperature = 0.01
+            >>> gumbel_softmax = F.gumbel_softmax(logits, temperature)
+            >>> print(gumbel_softmax)
+            >>> #out's value is as follows:
+                [[0.00000001, 1.        , 0.00000000, 0.00000000, 0.00000006, 0.00000000],
+                 [0.00000000, 0.00000000, 0.00000000, 0.00000000, 0.00000000, 1.        ],
+                 [0.00000062, 0.00000000, 0.00000000, 0.00000000, 0.00000000, 0.99999940],
+                 [0.00000000, 0.00000000, 0.00000000, 0.00001258, 0.99998736, 0.00000000]]
```
Comment (sunzhongkai588) [inline python/paddle/nn/functional/activation.py]: 像这类代码中带有随机性的，如 `paddle.randn([4, 6])`，需要设置随机种子，如加一行 `paddle.seed(2023)`，以此确保结果可复现，能够让CI进行验证
```suggestion
            >>> paddle.seed(2023)
            >>> logits = paddle.randn([4, 6])
            >>> temperature = 0.01
            >>> gumbel_softmax = F.gumbel_softmax(logits, temperature)
            >>> print(gumbel_softmax)
            [[0.00000001, 1.        , 0.00000000, 0.00000000, 0.00000006, 0.00000000],
             [0.00000000, 0.00000000, 0.00000000, 0.00000000, 0.00000000, 1.        ],
             [0.00000062, 0.00000000, 0.00000000, 0.00000000, 0.00000000, 0.99999940],
             [0.00000000, 0.00000000, 0.00000000, 0.00001258, 0.99998736, 0.00000000]]
```

**Review by sunzhongkai588 — COMMENTED** [review]
有很多需要同意修改的地方，如

1. 如 `celu` 修改意见所示。代码示例如果有输出，应该加 `print()` 或者直接写变量名 `out` 。并且输出的内容的缩进应该和 `>>>` 对齐。

2. 如 `gumbel_softmax` 修改意见所示。代码中带有随机性的，如 `paddle.randn([4, 6])`，需要设置随机种子，如加一行 `paddle.seed(2023)`，以此确保结果可复现，能够让CI进行验证

**Comment by sunzhongkai588** [comment]
@yuchen202  请在修改时，务必仔细参考任务描述⚠️：https://github.com/PaddlePaddle/Paddle/issues/55629
image
除review意见外，还需要注意⚠️：
1. 修改完代码示例后，务必现在在本机进行验证，验证命令为 `convert-doctest doctest xxx.py`
2. PR描述需要按照任务描述里那样写，尽量写清晰

```
<!-- Demo: https://github.com/PaddlePaddle/Paddle/pull/24810 -->
### PR types
<!-- One of [ New features | Bug fixes | Function optimization | Performance optimization | Breaking changes | Others ] -->
Others

### PR changes
<!-- One of [ OPs | APIs | Docs | Others ] -->
Others

### Description
<!-- Describe what you’ve done -->

修改如下文件的示例代码，使其通过 `xdoctest` 检查：

- `XXX.py`
- `XXX.py`

预览：

- en（英文文档）: http://preview-paddle-pr-XXX.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/en/api/paddle/XXX_en.html
- zh（中文文档）: http://preview-paddle-pr-XXX.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/XXX_cn.html

### Related links
 
- #55629
- #55295

@sunzhongkai588 @SigureMo @megemini 
```
3. 之前的PR废弃了，就把它close了吧 https://github.com/PaddlePaddle/Paddle/pull/55933。 下次注意，同一个PR，每次修改后都提交在同一分支上，这样就不用每次修改都换PR

File: python/paddle/nn/functional/activation.py [hunk 3]
```diff
@@ -502,25 +507,25 @@ def prelu(x, weight, data_format="NCHW", name=None):
...
+            >>> import paddle
+            >>> import paddle.nn.functional as F
+
+            >>> data = paddle.to_tensor([[[[-2.0,  3.0, -4.0,  5.0],
+            >>>                    [ 3.0, -4.0,  5.0, -6.0],
+            >>>                    [-7.0, -8.0,  8.0,  9.0]],
+            >>>                   [[ 1.0, -2.0, -3.0,  4.0],
+            >>>                    [-5.0,  6.0,  7.0, -8.0],
+            >>>                    [ 6.0,  7.0,  8.0,  9.0]]]], dtype='float32')
+
```
Comment (megemini) [inline python/paddle/nn/functional/activation.py]: 这是用 convert-doctest 工具转换的？多行的代码应该用 `PS2(... )` 进行连接 ～

image

另外，为了美观，可以把方括号对齐一下～

File: python/paddle/nn/functional/activation.py [hunk 4]
```diff
@@ -1264,14 +1271,14 @@ def softshrink(x, threshold=0.5, name=None):
...
-            x = paddle.to_tensor([-0.9, -0.2, 0.1, 0.8])
-            out = F.softshrink(x)
-            print(out)
-            # Tensor(shape=[4], dtype=float32, place=Place(gpu:0), stop_gradient=True,
-            #        [-0.39999998,  0.        ,  0.        ,  0.30000001])
+            >>> x = paddle.to_tensor([-0.9, -0.2, 0.1, 0.8])
+            >>> out = F.softshrink(x)
+            >>> print(out)
+                Tensor(shape=[4], dtype=float32, place=Place(gpu:0), stop_gradient=True,
+                [-0.39999998,  0.        ,  0.        ,  0.30000001])
```
Comment (megemini) [inline python/paddle/nn/functional/activation.py]: 在本地运行的时候，可以先执行 `paddle.device.set_device('cpu')` ，让 Tensor 放到 cpu 上去执行～

File: python/paddle/nn/functional/common.py [hunk 5]
```diff
@@ -876,15 +874,14 @@ def upsample(
...
-            upsample_out = paddle.nn.Upsample(size=[12,12])
-
-            output = upsample_out(x=input_data)
-            print(output.shape)
-            # [2L, 3L, 12L, 12L]
+            >>> input_data = paddle.randn(shape=(2,3,6,10)).astype(paddle.float32)
+            >>> upsample_out = paddle.nn.Upsample(size=[12,12])
+            >>> output = upsample_out(x=input_data)
+            >>> print(output.shape)
+                [2L, 3L, 12L, 12L]
```
Comment (megemini) [inline python/paddle/nn/functional/common.py]: 这次的任务不只是把原来的示例代码改一下，还要保证新的示例运行正确 ～

所以，每次改完之后，可以在本地运行一下，看看结果一不一样。

image

验证结果，既可以用 python 自己的交互工具，也可以用 ipython。

如上图所示，可以把改完的代码直接复制到 ipython 中，就可以看到运行结果了 ～

File: python/paddle/nn/functional/common.py [hunk 6]
```diff
@@ -2081,67 +2078,67 @@ class centers and the shape of sampled_class_center will be [num_positive_class_
...
-        #       [0, 2, 4, 8, 9, 3])
-
-        # rank 1 output:
-        #Tensor(shape=[20], dtype=int64, place=CUDAPlace(1), stop_gradient=True,
-        #       [10, 17, 15, 11, 9 , 12, 18, 18, 17, 18, 19, 2 , 8 , 13, 11, 13, 9 , 10, 0 , 4 ])
-        #Tensor(shape=[20], dtype=int64, place=CUDAPlace(1), stop_gradient=True,
-        #       [6 , 11, 10, 7 , 4 , 8 , 12, 12, 11, 12, 13, 1 , 3 , 9 , 7 , 9 , 4 , 6 , 0 , 2 ])
-        #Tensor(shape=[7], dtype=int64, place=CUDAPlace(1), stop_gradient=True,
-        #       [0, 1, 2, 3, 5, 7, 8])
+        >>> # required: distributed
```
Comment (megemini) [inline python/paddle/nn/functional/common.py:2107]: 对于需要 gpu、distributed 的环境，这里可以改为 ：

``` python
>>> # doctest: +REQUIRES(env: DISTRIBUTED)
```

**Comment by megemini** [comment]
@yuchen202 把 PR 的标题和描述根据 issue 的规定改一下吧 ～ 如果有什么问题，可以在这里 @ 大家 ～

啥时候准备好了 review 说一下哈 ～ ：）

**Comment by yuchen202** [comment]
https://github.com/PaddlePaddle/Paddle/pull/56440
因为不会在原来的PR改了。。。

**Comment by luotao1** [comment]
> 因为不会在原来的PR改了

加油，这个得学习一下。不然新PR是没法直观看到和老PR的diff的，不利于审核人review。

**Comment by megemini** [comment]
> #56440 因为不会在原来的PR改了。。。

只要在这个 PR 的 git branch 中修改并提交就可以了 ～

1. 切换到需要提交的分支
``` shell
$ git checkout xdoctest
```

2. 添加修改的文件
``` shell
$ git add XXX
```
`XXX` 不要用 `.` 代替，不然有可能加入一些无关的文件。

3. 用代码检查
``` shell
$ pre-commit
```
如果检查出问题，工具有可能帮助修改，此时，需要重复步骤 `2`。

4. 提交代码
``` shell
$ git commit -m "XXXXX"
```

5. 推送到github
``` shell
$ git push origin xdoctest
```

第 5 步之前最好还能同步一下 paddle 最新的代码库。

如果有新的 PR 要提，可以 
``` shell
$ git checkout -b XXX
```
新建一个分支。如果又要修改这个 PR 了，再用 
``` shell
$ git checkout xdoctest
```
切换回来就行。

具体的步骤可以参考：[文档贡献指南](https://www.paddlepaddle.org.cn/documentation/docs/zh/dev_guides/docs_contributing_guides_cn.html)

写的很详细 ～

另外，提个小建议，如果刚接触 git/github ，最好不要用一些专用的工具来操作，先把命令一条条的摸熟悉了，这样后面有问题好解决 ～

**Comment by SigureMo** [comment]
建议前期先每个 PR 改一个文件吧

啊我是说先提一个 PR，并且改完 merge 了再提新的 PR，不是说一次性提十多个 PR，那同样没啥意义，熟悉了之后再一个 PR 改多个文件
